diff --git a/drivers/pinctrl/pinctrl-single.c b/drivers/pinctrl/pinctrl-single.c
index 8a70b7c4f7ee..0391fb57a91c 100644
--- a/drivers/pinctrl/pinctrl-single.c
+++ b/drivers/pinctrl/pinctrl-single.c
@@ -44,6 +44,8 @@
 
 #ifdef CONFIG_SOC_SPACEMIT_K1X
 #define EDGE_CLEAR			6
+#define EDGE_FALL_EN			5
+#define EDGE_RISE_EN			4
 #endif
 
 /**
@@ -1825,7 +1827,9 @@ static int pcs_probe(struct platform_device *pdev)
 	struct resource *res;
 	struct pcs_device *pcs;
 	const struct pcs_soc_data *soc;
-	int ret;
+	int ret, i;
+	u32 regval;
+	void __iomem *base;
 
 	soc = of_device_get_match_data(&pdev->dev);
 	if (WARN_ON(!soc))
@@ -1976,6 +1980,16 @@ static int pcs_probe(struct platform_device *pdev)
 		pcs->desc.confops = &pcs_pinconf_ops;
 	pcs->desc.owner = THIS_MODULE;
 
+#ifdef CONFIG_SOC_SPACEMIT_K1X
+	for (i = 0, base = pcs->base; i < pcs->size; i += 4, base += 4) {
+		regval = pcs->read((void __iomem *)base);
+		regval |= (1 << EDGE_CLEAR);
+		regval &= ~(1 << EDGE_FALL_EN);
+		regval &= ~(1 << EDGE_RISE_EN);
+		pcs->write(regval, (void __iomem *)base);
+	}
+#endif
+
 	ret = pcs_allocate_pin_table(pcs);
 	if (ret < 0)
 		goto free;
